src/explorer.api/Exploration/ExplorationLauncher.cs

```diff
@@ -0,0 +1,110 @@
+namespace Explorer.Api
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Aircloak.JsonApi;
+    using Diffix;
+    using Explorer;
+    using Explorer.Api.Authentication;
+    using Explorer.Components;
+    using Lamar;
+
+    public class ExplorationLauncher
+    {
+        private readonly IContainer rootContainer;
+
+        public ExplorationLauncher(IContainer rootContainer)
+        {
+            this.rootContainer = rootContainer;
+        }
+
+        /// <summary>
+        /// Configures and runs an Exploration.
+        /// </summary>
+        /// <param name="scope">This should be a fresh nested scope based off the main container.</param>
+        /// <param name="data">The params of the current explorer api request.</param>
+        /// <param name="ct">A cancellation token that will be passed to subtasks.</param>
+        /// <returns>A task that represents the configured Exploration.</returns>
+        public static async Task Explore(INestedContainer scope, Models.ExploreParams data, CancellationToken ct)
+        {
+            // Register the authentication token for this scope.
+            if (scope.GetInstance<IAircloakAuthenticationProvider>() is ExplorerApiAuthProvider auth)
+            {
+                auth.RegisterApiKey(data.ApiKey);
+            }
+
+            // Create the Context and Connection objects for this exploration.
+            var ctx = await scope.GetInstance<ContextBuilder>().Build(data);
+            var conn = scope.GetInstance<AircloakConnectionBuilder>().Build(data.DataSourceName, ct);
```

Comment:
It's true that to benefit fully from DI we could define factory interfaces for each of these, however at the moment there isn't much benefit since there is nowhere where we might want to use a different `ContextBuilder` or `AircloakConnectionBuilder`. The use case is fairly narrow (take the request data and use it to build the required instance). The only place I think it might be useful would be in testing. 

Also note I have made some changes to this in the `daniel/explorer-components-v2` branch (I think the context and connection should be passed as parameters to the exploration rather than the creation of these being the responsibility of the launcher). If we introduce an interface, I would rather do it on top this later branch.

Reply:
This is not very clear:
> Also note I have made some changes to this in the daniel/explorer-components-v2 branch (I think the context and connection should be passed as parameters to the exploration rather than the creation of these being the responsibility of the launcher). If we introduce an interface, I would rather do it on top this later branch.

Reply:
I meant `daniel/dynamic-air-backend` branch, sorry... 

In that branch I changed the signature of the explorer launch method to take an `ExplorerContext` and a `DConnection` instead of the `ExploreParams`. This way I could move the `ExplorationLauncher` to the explorer project. 

So the logic of how to build the ExplorerContext and DConnection will reside in the `ExploreController`, and tying together the components and running them is in the explorer project.